Data/Table/Operations.idr

```diff
@@ -0,0 +1,45 @@
+module Data.Table.Operations
+
+import public Data.Table.Column
+import public Data.Table.Data
+import public Data.Table.Record
+import public Data.Table.Row
+import public Data.Table.Schema
+
+public export
+(|*|) : Table schema1 -> Table schema2 -> Table (schema1 |+| schema2)
+[<] |*| table2 = [<]
+(table1 :< rec) |*| table2 = (table1 |*| table2) ++ (rec |*| table2)
+
+public export
+0
+crossJoinHasRows :
+     (table1 : Table schema1)
+  -> (hasRows1 : table1 `HasRows` n1)
+  => (table2 : Table schema2)
+  -> (hasRows2 : table2 `HasRows` n2)
+  => (table1 |*| table2) `HasRows` n1 * n2
+crossJoinHasRows [<] table2 {hasRows1 = EmptyTable} = EmptyTable
+crossJoinHasRows (table1 :< rec) table2 {hasRows1 = SnocTable hasRows1}
+  = let u1 = crossJoinHasRows table1 table2
+        u2 = crossJoinHasRows rec    table2
+    in (vcatHasRows (table1 |*| table2) (rec |*| table2))
+    `ByProof` (plusCommutative _ _)
+
+namespace Record
+  public export
+  (|*|) : Record schema1 -> Frame schema2 n -> Frame (schema1 |+| schema2) n
+  rec |*| frame = (rec |*| frame.fst) `Element`
```

Comment:
There are some utility functions in `Data.Table.Row.Frame`, including a `%hint`, which should simplify this to `frame $ rec |*| table frm`.

That said, I want to discourage use of `Frame`s, as otherwise we'd end up with two copies of every single function. So I don't think we should include the `Frame` variants of these functions.